### ISSUE TITLE
ds.levels and ds.summary

### ISSUE THREAD

**Issue by tombisho** [datashield/dsBetaTest#37]
I think we need to make a new version of ds.levels that includes the disclosure trap for number of individuals in a group. If we have a numeric variable on the server side, this can be converted to a factor using ds.asFactor. Then the levels are the individual values, which can simply be returned with ds.levels.

Actually thinking about it, ds.summary will also return the level labels with NA in this scenario, I think, and should be modified not to return the labels.

**Comment by davraam** [comment]
> I think we need to make a new version of ds.levels that includes the disclosure trap for number of individuals in a group. If we have a numeric variable on the server side, this can be converted to a factor using ds.asFactor. Then the levels are the individual values, which can simply be returned with ds.levels.
> 
> Actually thinking about it, ds.summary will also return the level labels with NA in this scenario, I think, and should be modified not to return the labels.

Yes, that's right we have to add a disclosure trap that checks the density of each level when converting a numeric to factor.

**Comment by tombisho** [comment]
Great. Is this already on a roadmap or development plan?

**Comment by davraam** [comment]
Let's add this on our roadmap. The check of the levels' density is also neccessary in some of the graphical functions. Thanks Tom for the suggestion!
